docs/developer_guides/how-guarantee-ts-works.md

```diff
@@ -0,0 +1,80 @@
+# Guarantee Timestamp in Search Requests
+
+[中文文档](./how-guarantee-ts-works-cn.md)
+
+Users and developers may feel confused by the so many parameters in the search requests, especially for those developers
+who develop sdk client for Milvus.
+
+This document will explain a special parameter in the search request-"Guarantee Timestamp", hereinafter referred to as
+"GuaranteeTs".
+
+## Milvus Clock Mechanism
+
+Like most distributed systems, Milvus will assign a timestamp to each record that enters the system.
+
+At the same time, Milvus is a system that separates storage and computing. 
+
+The DataNodes handle the data persistence load, and the data will eventually be flushed to MinIO/S3 or other distributed
+object storage.
+
+QueryNodes handle the computing(read) requests. The reading link will process two types of data at the same time--batch 
+data and streaming data.
+
+Among them, the batch data will no longer be changed, and the search request will see all the data in the batch data.
+
+QueryNodes and DataNodes consume user insert requests through the same subscription mechanism, which constitute the
+streaming data.
+
+Due to the network delays, QueryNodes often do not hold all the latest streaming data. Without other control mechanism,
```

Comment:
lose some data is horrible,  invisible is more accurate

Reply:
OK, got it.